### PR TITLE
Fix #5840: PanelGrid Support row and column on responsive

### LINKED ISSUE #5840: PanelGrid: No way to manage style or identifiers of panelgrid cells except for tabular mode
Responsive panelgrid does not currently allow specific cells within the grid to be identified or styled uniquely. This capability only exists with the layout='tabular' mode.

This pull requests now enables row and column components to be used with layout='grid' and column components to be used with layout='flex'.

The usefulness of this can be seen in the context of legacy tabular mode which allowed for rows to be highlighted differently based on odd or even colors. In the context of responsive design there is still a need to control the styling of rows and cells within the grid and this pull request enables that.

## 1) Environment
- PrimeFaces version: All
- Does it work on the newest released PrimeFaces version? No
- Does it work on the newest sources in GitHub? No
- Application server + version: Irrelevant
- Affected browsers: Irrelevant

## 2) Expected behavior

Need a method to add style, styleClass, and ID attributes to DIVs created

## 3) Actual behavior

Feature Request, Not a bug

## 4) Steps to reproduce

Feature Request, Not a bug

## 5) Sample XHTML

```xml
<p:panelGrid columns="2" columnClasses="ui-grid-col-4,ui-grid-col-8" layout="grid">

<p:row styleClass="ui-lets-make-this-special">
     User Account
</p:row>

<p:outputLabel for="field01" value="First Name"/>
<p:inputText id="field01" value="#{user.firstname}"/>

<p:column id="LastNameLabel">
    <p:outputLabel for="field02" value="Last Name"/>
</p:column>
<p:inputText id="field02" value="#{user.lastname}"/>

<p:row style="background-color: lightgray;">
     Other
</p:row>

<p:outputLabel for="field03" value="Cats Name"/>
<p:inputText id="field03" value="#{user.catsname}"/>

<p:outputLabel for="field04" value="Vet Number"/>
<p:inputText id="field04" value="#{user.vetNumber}"/>

<p:row id="lastRow" styleClass="my-header-style-class">
<p:column id="lastRowLeft"><p:outputLabel for="field05" value="Some Title"/></p:column>
<p:column id="lastRowRight"><p:inputText id="field05" value="#{user.someTitle}"/></p:column>
</p:row>

</p:panelGrid>
```

## 6) Sample bean

Not a bug

### PR DESCRIPTION
**Pull request by tim12s**
Responsive panelgrid does not currently allow specific cells within the grid to be identified or styled uniquely.  This capability only exists with the layout='tabular' mode.

This pull requests now enables row and column components to be used with layout='grid' and column components to be used with layout='flex'.

The usefulness of this can be seen in the context of legacy tabular mode which allowed for rows to be highlighted differently based on odd or even colors.  In the context of responsive design there is still a need to control the styling of rows and cells within the grid and this pull request enables that.

### CODE REVIEW

**Comment by tandraschko** [comment]
Hi,
could you please create a issue before (PRs are always based on issues) and also put your description there?

**Comment by melloware** [comment]
@tim12s Please create an issue in the Issue section so we can link this PR.  

Also you have a CheckStyle violation so the build is failing.  Run `mvn clean package` to see the error locally.

**Comment by tim12s** [comment]
Hi.  The checkstyle has been resolved and I have removed the documentation changes I made for version 8 documentation as I see the milestone has been set for version 9.

Once this is in, I will look at updating the showcase.

**Review by melloware — APPROVED** [review]

**Comment by melloware** [comment]
@christophs78 I think you were the last one in here to add "flex" support do you mind giving this a review?

**Comment by christophs78** [comment]
We need to keep panelgrid.md for 8.0. Can you restore this one please?
(We have to keep old docu-versions for people who still use them.)

File: docs/9_0/components/panelgrid.md [hunk 1]
```diff
@@ -103,6 +103,56 @@ However, you may find your labels not lining up in horizontal forms.  You can us
...
+The respective cell style will be utilized from the provided `columnClasses` and from any `styleClass` on the
+component.  If more components are contained within the row component then the respective `columnClasses` will
+be utilized based `columnClasses[column%columns]`.
+
+
+```xml
+<p:panelGrid 
+        columns="2" 
+        layout="grid"
+        columnClasses="ui-grid-col-8,ui-grid-col-4,ui-grid-col-8,ui-grid-col-4">
```
Comment (christophs78) [inline docs/9_0/components/panelgrid.md:125]: You set columns to 2 but add 4 column-classes. Is this by intention?

**Comment by christophs78** [comment]
@tim12s : First off all thanks for your work you put into this PR.
Did you check it against https://www.primefaces.org/showcase/ui/panel/panelGrid.xhtml to avoid regressions? I will try to run some tests myself within the next few days.

**Comment by tim12s** [comment]
I was considering showing an example of a row with more than 2 columns embedded in the row.  These columns would then require a column class.

I might make a small change to the indexing internal to use all the column classes if necessary.

The current indexing is using index % columns but within a row i am thinking of using index % columnClasses.length.

**Comment by tim12s** [comment]
I will also add an example with ui-float-label as the current method of float label with a span results in the generation of a UIInstruction which breaks the column model.

This is corrected in this pull request by allowing a ui-float-label inputText to be wrapped in a p:column.

**Comment by christophs78** [comment]
„Green flag“ from my side. Your code seems to be fine.

Just had been a bit confused by the example you added to the documentation. This example mixes „things“ together you won´t mix in a real world usecase?

As you wrote before a ShowCase-PR is welcome which add´s some examples. This should make it easier to get into it. And it help´s ourself to avoid regressions in future PR´s.

File: docs/9_0/components/panelgrid.md (hunk 1 above)
Comment (tim12s) [reply docs/9_0/components/panelgrid.md:125]: This conversation was resolved by a comment on the main discussion.


> I was considering showing an example of a row with more than 2 columns embedded in the row. These columns would then require a column class.
> 
> I might make a small change to the indexing internal to use all the column classes if necessary.
> 
> The current indexing is using index % columns but within a row i am thinking of using index % columnClasses.length.

**Comment by melloware** [comment]
@tim12s Are you happy with this PR?  I am ready to merge it so you can create your Showcase PR.

**Comment by melloware** [comment]
@christophs78 and I have both reviewed it and it looks good to me.

**Comment by tim12s** [comment]
Yes I am happy with it.

**Comment by tim12s** [comment]
If I have 1 line of code to submit must I log a new issue or create a pull request and link it to the original issue?

**Comment by christophs78** [comment]
IMO new PR linked to the original issue. (Because we still work on 9.0 and i expect it´s a minor improvement to what you did already.)

**Comment by melloware** [comment]
Yep create a new PR but use the old issue number.

**Pull request merged**